src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplate.cs

```diff
@@ -78,13 +78,25 @@ public static BindingTemplate FromString(string input)
         /// a dictionary.
         /// </summary>
         /// <param name="parameters">Dictionary providing parameter values.</param>
+        /// <param name="ignoreCase">True if matching should be case insensitive.</param>
         /// <returns>Resolved string if succeeded.</returns>
         /// <exception cref="InvalidOperationException">Thrown when required parameter value is not available.
         /// </exception>
-        public string Bind(IReadOnlyDictionary<string, string> parameters)
+        public string Bind(IReadOnlyDictionary<string, string> parameters, bool ignoreCase = false)
         {
             StringBuilder builder = new StringBuilder();
 
+            if (ignoreCase && parameters != null)
+            {
+                // convert to a case insensitive dictionary
+                var caseInsensitive = new Dictionary<string, string>(parameters.Count, StringComparer.OrdinalIgnoreCase);
```

Comment:
Can we add the ignoreCase param to public static BindingTemplate FromString(string input) as well?
That's what we are using mostly in  extentions